src/renderer/stores/project.ts

```diff
@@ -507,7 +512,27 @@ export const useCreateEdge = () => {
       );
     }
 
-    setEdges((edges) => addEdge(connection, edges));
+    setEdges((edges) => {
+      const edgeExists = edges.find(
+        (edge) =>
+          edge.source === connection.source &&
+          edge.target === connection.target &&
+          edge.sourceHandle === connection.sourceHandle &&
+          edge.targetHandle === connection.targetHandle,
+      );
+      if (edgeExists || !connection.source || !connection.target) {
+        return edges;
+      }
+      const newEdge: Edge<EdgeData> = {
+        id: `${connection.source}->${connection.target}`,
```

Comment:
this id is not unique per edge, if you have two edges from block A to block B then both of them will have the same id, it's probably better to use a uuid (or do something like ``${connection.source}->${connection.target} ${(connection.sourceHandle)->(connection.targetHandle)}``

Reply:
Nice catch @39bytes . Fixed.